bundles/org.openhab.binding.nzwateralerts/src/main/java/org/openhab/binding/nzwateralerts/internal/handler/NZWaterAlertsHandler.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nzwateralerts.internal.handler;
+
+import static org.openhab.binding.nzwateralerts.internal.NZWaterAlertsBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.nzwateralerts.internal.NZWaterAlertsConfiguration;
+import org.openhab.binding.nzwateralerts.internal.binder.NZWaterAlertsBinder;
+import org.openhab.binding.nzwateralerts.internal.binder.NZWaterAlertsBinderListener;
+
+/**
+ * The {@link NZWaterAlertsHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+@NonNullByDefault
+public class NZWaterAlertsHandler extends BaseThingHandler implements NZWaterAlertsBinderListener {
+
+    private @Nullable NZWaterAlertsConfiguration config = null;
+    private HttpClient httpClient;
+    private @Nullable NZWaterAlertsBinder binder = null;
+
+    public NZWaterAlertsHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (CHANNEL_ALERTLEVEL.equals(channelUID.getId())) {
+            if (command instanceof RefreshType) {
+                if (binder != null)
+                    binder.update();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(NZWaterAlertsConfiguration.class);
+
+        binder = new NZWaterAlertsBinder(httpClient, config, scheduler);
```

Comment:
See above store also to a local variable and work on that. Same for below.
```suggestion
        NZWaterAlertsBinder localbinder = binder = new NZWaterAlertsBinder(httpClient, config, scheduler);
```